### PR TITLE
Extend GithubActions basic workflow

### PR DESCRIPTION
**Pull request by PatKamin**
Add new configurations like os, compiler, build type, etc. based on the GithubActions main workflow's configuration in the Unified Runtime.

### CODE REVIEW

File: .github/workflows/basic.yml [hunk 1]
```diff
@@ -4,79 +4,120 @@ on: [push, pull_request]
...
+          -DUMF_DEVELOPER_MODE=ON
+          -DUMF_ENABLE_POOL_TRACKING=ON
+
+      - name: Run code-style check     
+        run: |
+          cmake --build ${{github.workspace}}/build --target clang-format-check
+
+      - name: Build UMF
+        run: |
+          cmake --build ${{github.workspace}}/build -j 2
```
Comment (kswiecicki) [inline .github/workflows/basic.yml:123]: `-j $(nproc)`

File: .github/workflows/basic.yml [hunk 2]
```diff
@@ -4,79 +4,120 @@ on: [push, pull_request]
...
           sudo apt-get install -y clang git cmake python3-pip
+      
+      - name: Install g++-7
+        if: matrix.compiler.cxx == 'g++-7'
+        run: |
+          sudo apt-get install -y ${{matrix.compiler.cxx}}
 
       - name: Install pip packages
         run: |
           pip3 install clang-format==15.0.7
```
Comment (bratpiorka) [inline .github/workflows/basic.yml:40]: you don't check coding style here so you don't need to install clang-format

File: .github/workflows/basic.yml [hunk 3]
```diff
@@ -4,79 +4,120 @@ on: [push, pull_request]
...
-
-    - name: Build all
-      run: cmake --build ${{github.workspace}}/build --config ${{matrix.build_type}} -j 2
-
-    - name: Test
-      working-directory: ${{github.workspace}}/build
-      run: ctest -C ${{matrix.build_type}} --output-on-failure
+      - uses: actions/checkout@v3
+
+      - name: Install prerequisites
```
Comment (bratpiorka) [inline .github/workflows/basic.yml:75]: you don't use clang-format here

**Review by bratpiorka — CHANGES_REQUESTED** [review]

File: .github/workflows/basic.yml [hunk 4]
```diff
@@ -4,79 +4,120 @@ on: [push, pull_request]
...
+          cmake
+          -B${{github.workspace}}/build
+          -DCMAKE_C_COMPILER=${{matrix.compiler.c}}
+          -DCMAKE_CXX_COMPILER=${{matrix.compiler.cxx}}
+          -DUMF_FORMAT_CODE_STYLE=OFF
+          -DUMF_DEVELOPER_MODE=ON
+          ${{matrix.pool_tracking}}
+
+      - name: Build UMF
+        run: cmake --build ${{github.workspace}}/build --config ${{matrix.build_type}} -j 2
```
[the remark this answers is not in the document]
Comment (PatKamin) [reply .github/workflows/basic.yml:89]: Done

File: .github/workflows/basic.yml [hunk 5]
```diff
@@ -4,79 +4,120 @@ on: [push, pull_request]
...
+    strategy:
+        matrix:
+          os: ['macos-12']
+    runs-on: ${{matrix.os}}
+
+    steps:
+      - uses: actions/checkout@v3
+
+      - name: Install prerequisites
+        run: python3 -m pip install clang-format==15.0.7
```
[the remark this answers is not in the document]
Comment (PatKamin) [reply .github/workflows/basic.yml:106]: Done

**Review by bratpiorka — APPROVED** [review]

**Review by kswiecicki — APPROVED** [review]

**Pull request merged**